segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/ContainerRecoveryUtils.java

```diff
@@ -246,6 +254,112 @@ private static void validateContainerIds(Map<Integer, DebugStreamSegmentContaine
+        for (val backUpMetadataSegmentEntry : backUpMetadataSegments.entrySet()) {
+            // Get the name of original metadata segment
+            val metadataSegment = NameUtils.getMetadataSegmentName(backUpMetadataSegmentEntry.getKey());
+
+            // Get the name of back up metadata segment
+            val backUpMetadataSegment = backUpMetadataSegmentEntry.getValue();
+
+            // Get the new container assignment for back up metadata segment
+            val containerForBackUpMetadataSegment = containersMap.get(segToConMapper.getContainerId(
+                    backUpMetadataSegment));
+            log.info("Back up container metadata segment name: {} and its container id: {}", backUpMetadataSegment,
+                    containerForBackUpMetadataSegment.getId());
+
+            // Get the iterator to iterate through all segments in the back up metadata segment
+            val tableExtension = containerForBackUpMetadataSegment.getExtension(ContainerTableExtension.class);
+            val entryIterator = tableExtension.entryIterator(backUpMetadataSegment, args)
+                    .get(TIMEOUT.toMillis(), TimeUnit.MILLISECONDS);
+
+            ArrayList<CompletableFuture<Void>> futures = new ArrayList<>();
+
+            // Iterating through all segments in the back up metadata segment
+            entryIterator.forEachRemaining(item -> {
+                for (val entry : item.getEntries()) {
+
+                    val segmentInfo = MetadataStore.SegmentInfo.deserialize(entry.getValue());
+                    val properties = segmentInfo.getProperties();
+
+                    // skip, if this is original metadata segment
+                    if (properties.getName().equals(metadataSegment)) {
+                        continue;
+                    }
+
+                    // Get the attributes for the current segment
+                    List<AttributeUpdate> attributeUpdates = properties.getAttributes().entrySet().stream()
+                            .map(e -> new AttributeUpdate(e.getKey(), AttributeUpdateType.Replace, e.getValue()))
+                            .collect(Collectors.toList());
+                    log.info("Segment Name: {} Attributes Updates: {}", properties.getName(), attributeUpdates);
+
+                    // Get the container for the current segment
+                    val container = containersMap.get(segToConMapper.getContainerId(
+                            properties.getName()));
+
+                    // Update attributes for the current segment
+                    futures.add(Futures.exceptionallyExpecting(
+                            container.updateAttributes(properties.getName(), attributeUpdates, TIMEOUT),
+                            ex -> ex instanceof StreamSegmentNotExistsException, null));
+                }
+
+                // Waiting for update attributes for all segments in each back up metadata segment.
+                Futures.allOf(futures).join();
```

Comment:
What I meant is to add this outside of the `entryIterator.forEachRemaining` but still withinin the outer for loop.

And when you wait on it, use `.get(with timeout)` so we don't end up waiting forever.

In that case we also need to make sure the Timeout is something reasonable. If we set it to 5s, then that may not be sufficient to restore all the segments in a container (there could be millions of them).

Reply:
Thanks for telling that. I am now, taking timeOut as a parameter to the method. The tests which use this method, will pass their timeOut values here. 
I will do similar thing for other methods in this class like `recoverAllSegments` & `copySegment` in this class.